fe/lib/components/LocationSuggest/LocationSuggest.tsx

```diff
@@ -139,6 +154,23 @@ export const LocationSuggest = forwardRef<HTMLInputElement, Props>(
       getLocationSuggest,
     ]);
 
+    useEffect(() => {
+      if (initialValue) {
+        getLocation({
+          variables: {
+            id: initialValue,
+          },
+        });
+      }
+    }, [initialValue, getLocation]);
+
+    useEffect(() => {
+      setInitialLocation(locationData?.location ?? undefined);
+      if (locationData?.location) {
+        setSelectedLocationId(locationData?.location.id.value);
+      }
+    }, [locationData]);
```

Comment:
Can/should we collapse this into one effect? e.g.

```typescript
const loadInitialLocation = useCallback(async () => {
  if (!initialValue || initialLocation?.id.value === initialValue) {
    return;
  }

  const { data } = await client.query();

  if (!data.location) {
    return;
  }

  setInitialLocation(data.location);

  // TODO: do we want this if?
  // It would avoid overriding the user's selection if they are somehow super quick.
  if (!selectedLocationId) {
    setSelectedLocationId(data.location.id.value);
  }
}, [initialValue]);

useEffect(() => {
  loadInitialLocation();
}, [blah]);
```